### PR TITLE
fix(chat): don't misreport missing worktree as missing CLI

### PR DESCRIPTION
**Pull request by jamesbrink**
## Why

A deleted worktree directory was surfacing in the UI as **“Claude CLI is not installed”** with a full-screen install-options modal popping on every chat-send. The CLI was fine — `Command::spawn()` collapses both `chdir(current_dir)` and `execvp(program)` failures into the same `ErrorKind::NotFound`, so our `map_spawn_err` blindly mapped *every* spawn `NotFound` to the missing-CLI sentinel. The actual cause (worktree dir gone) became invisible, the modal blocked workflow, and the recovery action (Archive / Recreate) was nowhere.

Two underlying issues, one symptom:
1. **Misclassification**: a missing `current_dir` lied about the binary.
2. **Hostile UX**: even when the CLI is genuinely missing, popping a full-screen modal on every chat send turns a recoverable error into a hard blocker.

### Root-cause fix — disambiguate spawn `NotFound`

- New sentinel `MISSING_CWD:<path>` in `src/missing_cli.rs` plus a `precheck_cwd()` helper. Every claude / plugin spawn site that calls `current_dir()` now pre-validates the directory exists before spawning, so the residual `NotFound` from `spawn()` is unambiguously about the binary.
- Sites updated: `agent::session::start`, `agent::process::run_turn`, `agent::naming::generate_branch_name` and `generate_session_name`, `plugin_runtime::host_api::host_exec`. The plugin-marketplace path (`plugin::run_claude_plugin_command`) already falls back to `$HOME` when the repo path is gone, so it didn't need the precheck.
- Tauri layer (`src-tauri/src/missing_cli.rs`) recognizes the new sentinel and emits a separate `missing-worktree` event with the path; the existing `missing-dependency` event is unchanged.

### Modal is no longer a blocker

- App.tsx listener now calls `reportMissingCli(payload)` instead of unconditionally opening the modal.
- The store decides per-tool: **first** event for a tool auto-opens (so non-chat surfaces — auth, repository, SCM, plugin settings — keep their direct-modal UX). After the user closes the modal for a tool, subsequent events for that same tool only refresh the cache; the modal stays closed. Inline “View install options →” link in `ChatErrorBanner` always reopens the modal AND clears the dismissal — explicit user action overrides snooze.
- Different tools dismiss independently — dismissing the `claude` modal in chat doesn't silence the `gh` modal coming from a future SCM sign-in attempt.

### Missing-worktree recovery surface

- New `missing-worktree` Tauri event surfaces the deleted worktree path.
- `ChatErrorBanner` (extracted from `ChatPanel.tsx`, which is in the god-file list) detects the missing-worktree friendly message and renders **Archive workspace** + **Recreate worktree** buttons backed by the existing `archive_workspace` / `restore_workspace` Tauri commands. Archive passes `skipScript: true` because the worktree is already gone — any repo `archive_script` that would chdir into it would just fail.
- Archive from this banner deselects the workspace (matches the sidebar archive button), dropping the user back to the “Start a workspace” empty state instead of leaving a stale chat tab open.

### Resume-fallback: respect structural failures

- The chat-send resume-fallback used to retry blindly with a fresh session even when the failure was structural (CLI gone, worktree gone) — wasted spawn + a duplicate event.
- `start_persistent` now returns the raw sentinel; each terminal `Err` arm calls `handle_err` itself, so event emission happens exactly once at the actual decision-to-fail. Sentinel-typed resume failures short-circuit instead of retrying.

### Shared workspace lifecycle hook

- New `useWorkspaceLifecycle` hook centralizes the optimistic-update → backend call → reconcile-or-rollback dance for archive / restore. ChatErrorBanner consumes it; the Sidebar's existing `handleArchive` keeps its archive-script confirmation flow but can be simplified to use the hook in a follow-up.

## Tests

- **Rust** (`cargo test -p claudette -p claudette-server -p claudette-cli --all-features`): 4 new unit tests in `src/missing_cli.rs` covering `precheck_cwd` (existing dir / missing dir / regular file), the `MISSING_CWD` sentinel parser, and the `is_sentinel` helper. Updated Tauri-side `friendly_message` test for the simplified wording.
- **Frontend** (`bun run test`): 1808 tests pass. New coverage:
  - 8 tests in `useAppStore.missingCli.test.ts` covering auto-open / dismissal / cross-tool independence / inline-link reopen.
  - 8 tests in `useWorkspaceLifecycle.test.tsx` covering optimistic update, deselect-when-selected, no-deselect-when-other-selected, hard-delete removal, rollback on failure (including the “user navigated away mid-flight” edge case), restore happy and sad paths, and `skipScript` forwarding.
  - 5 tests in `ChatErrorBanner.test.tsx` covering plain-error, missing-CLI inline link, missing-worktree archive (success + failure), and recreate.

## Commits in this PR

1. **`fix(chat): don't misreport missing worktree as missing CLI`** — root-cause fix at every spawn site, new `ChatErrorBanner` with inline link + recovery actions, Tauri-side event split.
2. **`fix(chat): archive from missing-worktree banner deselects workspace`** — `useWorkspaceLifecycle` hook so the banner's Archive matches the sidebar's UX.
3. **`fix(chat): address codex review on missing-cli/cwd handling`** — addresses two P2 findings from the codex peer review: non-chat surfaces regressed (fixed via per-tool dismissal model) and the structural-error resume guard was dead code (fixed by hoisting `handle_err` out of `start_persistent`).

## Reproducer

1. Create a workspace.
2. `rm -rf` its worktree directory while Claudette is running.
3. Send a chat message.

**Before**: full-screen “Claude CLI not installed” modal, friendly message in chat says “see install options dialog”, no recovery path. Modal re-pops on every send.

**After**: chat banner reads `Workspace directory is missing: /path/to/worktree. The worktree was deleted or moved — recreate it, or archive this workspace.` with **Archive workspace** and **Recreate worktree** buttons. Clicking Archive deselects and drops to the empty state.

## Docs

No new user-facing docs page — this is a bugfix to an existing error path. The change to friendly-message wording is internal contract between Rust and the frontend banner. Calling out here so the omission is intentional.

### CODE REVIEW

File: src/missing_cli.rs [hunk 1]
```diff
@@ -62,6 +96,43 @@ pub fn map_spawn_err(
...
+/// `current_dir(...)` — see the module docs for why.
+pub fn precheck_cwd(path: &Path) -> Result<(), String> {
+    // `is_dir()` follows symlinks and returns false for broken symlinks,
+    // missing entries, and regular files alike — all of which would cause
+    // chdir to fail at spawn time.
+    if path.is_dir() {
+        Ok(())
+    } else {
+        Err(format_cwd_err(path))
+    }
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/missing_cli.rs:178]: Addressed in 0xe776d1 / f6687cab. Switched to `fs::symlink_metadata` (with a follow-up `fs::metadata` for symlink targets) and only emit `MISSING_CWD` for `ErrorKind::NotFound` and the path-exists-but-is-not-a-directory case (chdir would `ENOTDIR`, same recovery UX). Other error kinds — `PermissionDenied`, transient IO — return a non-sentinel `"Cannot stat working directory <path>: <err>"` so callers fall through to their default error handler instead of offering the wrong recovery actions. New `precheck_cwd_returns_non_sentinel_on_permission_denied` test on Unix uses chmod 0o000 on a parent dir to pin the behavior.

File: src/ui/src/hooks/useWorkspaceLifecycle.test.tsx [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    mountedRoots.length = 0;
+    mountedContainers.length = 0;
+  });
+
+  it("optimistically marks the workspace archived and deselects it when it was selected", async () => {
+    seedWorkspace();
+    stateRef.state.selectedWorkspaceId = "ws-1";
+    const { archive } = await mountHook();
+
+    const result = await act(async () => archive("ws-1"));
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/hooks/useWorkspaceLifecycle.test.tsx]: Addressed in 0xe776d1. Refactored to `let result; await act(async () => { result = await archive("ws-1"); })` — captures via outer binding. Note for posterity: React 18+ `act` *is* generic and forwards the callback's resolved value (the tests were passing as written), but the outer-binding form removes the ambiguity flagged here and reads more clearly. Added a header comment in `useWorkspaceLifecycle.test.tsx` explaining the convention.

File: src/ui/src/hooks/useWorkspaceLifecycle.test.tsx [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+  it("rolls back optimistic state and restores selection on backend failure", async () => {
+    seedWorkspace();
+    stateRef.state.selectedWorkspaceId = "ws-1";
+    const fail = new Error("boom");
+    tauriApi.archiveWorkspace.mockRejectedValueOnce(fail);
+    const { archive } = await mountHook();
+    const result = await act(async () => archive("ws-1"));
+    expect(result).toEqual({ ok: false, error: fail });
+    // Last updateWorkspace call should be the rollback to the snapshot.
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/hooks/useWorkspaceLifecycle.test.tsx:203]: Addressed in 0xe776d1 — same refactor as the previous comment (outer `let result` assigned inside the `act` callback).

File: src/ui/src/hooks/useWorkspaceLifecycle.test.tsx [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    mountedContainers.length = 0;
+  });
+
+  it("flips workspace back to Active with the path returned by the backend", async () => {
+    seedWorkspace({ status: "Archived", worktree_path: null });
+    tauriApi.restoreWorkspace.mockResolvedValueOnce("/abs/restored/lush-daisy");
+    const { restore } = await mountHook();
+    const result = await act(async () => restore("ws-1"));
+    expect(result).toEqual({ ok: true });
+    expect(stateRef.updateWorkspace).toHaveBeenCalledWith("ws-1", {
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/hooks/useWorkspaceLifecycle.test.tsx:275]: Addressed in 0xe776d1 — refactored to outer `let result` for the restore happy-path test as well.

File: src/ui/src/hooks/useWorkspaceLifecycle.test.tsx [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+  });
+
+  it("returns an error result without mutating state when the backend fails", async () => {
+    seedWorkspace({ status: "Archived", worktree_path: null });
+    const fail = new Error("worktree add failed");
+    tauriApi.restoreWorkspace.mockRejectedValueOnce(fail);
+    const { restore } = await mountHook();
+    const result = await act(async () => restore("ws-1"));
+    expect(result).toEqual({ ok: false, error: fail });
+    expect(stateRef.updateWorkspace).not.toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/hooks/useWorkspaceLifecycle.test.tsx:291]: Addressed in 0xe776d1 — refactored to outer `let result` for the restore failure-path test as well. All four `act()` call sites in this file now use the same explicit pattern.

File: src/ui/src/components/chat/ChatErrorBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    // the Sidebar's archive-script confirmation modal is intentionally
+    // bypassed for this recovery surface (cf. useWorkspaceLifecycle docs).
+    const result = await archive(workspaceId, { skipScript: true });
+    setBusy(null);
+    if (result.ok) {
+      // `archive` already deselects the workspace, so the user lands on
+      // the "Start a workspace" empty state — same UX as clicking Archive
+      // in the sidebar context menu.
+      setLastMissingWorktree(null);
+      onRecovered?.();
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/ChatErrorBanner.tsx:87]: Addressed in 0xe776d1. Added a `mountedRef` (set true in a mount effect, cleared in cleanup) and gate post-await `setBusy` / `setActionError` calls on it. The success-path archive call still invokes `onRecovered?.()` unconditionally — the parent owns the error state and that is what actually matters when the banner is unmounted. Also belt-and-suspenders: a slow rollback that lands after the user navigated away no longer briefly resurrects the error UI.

File: src/ui/src/locales/en/chat.json [hunk 7]
```diff
@@ -38,6 +38,11 @@
   "voice_transcribing_with": "Transcribing with {{provider}}",
   "agent_question_placeholder": "Type a response...",
   "agent_question_submit": "Submit",
+  "missing_cli_inline_link": "View install options",
+  "missing_worktree_archive": "Archive workspace",
+  "missing_worktree_recreate": "Recreate worktree",
+  "missing_worktree_recreate_failed": "Failed to recreate worktree: {{error}}",
+  "missing_worktree_archive_failed": "Failed to archive workspace: {{error}}",
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/locales/en/chat.json:45]: Addressed in 0xe776d1. Added the 5 new keys to `es`, `ja`, `pt-BR`, and `zh-CN` `chat.json`, matching the existing tone in each locale (e.g. `"Archivar workspace"` / `"アーカイブ"` / `"Arquivar workspace"` / `"归档工作区"`).

File: src/missing_cli.rs [hunk 8]
```diff
@@ -62,6 +96,81 @@ pub fn map_spawn_err(
...
+/// Format the missing-cwd sentinel for a given path.
+pub fn format_cwd_err(path: &Path) -> String {
+    format!("{MISSING_CWD_PREFIX}{}", path.display())
+}
+
+/// If `err` carries the [`MISSING_CWD_PREFIX`] sentinel, return the path
+/// portion. Symmetric with [`parse_err`].
+pub fn parse_cwd_err(err: &str) -> Option<&str> {
+    let rest = err.strip_prefix(MISSING_CWD_PREFIX)?;
+    Some(rest.split_once(": ").map(|(p, _)| p).unwrap_or(rest))
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/missing_cli.rs]: Addressed in 0x72dd09. Removed the `split_once(": ")` from `parse_cwd_err` — the cwd sentinel format is strictly `MISSING_CWD:<path>` with no appended suffix, so the parser now returns the full remainder verbatim. New `parse_cwd_err_preserves_paths_containing_colon_space` test pins the contract using `/Users/me/Project: v2`. Doc comment notes that any future suffix needs an escaping or length-prefixed scheme rather than a raw delimiter that can collide with valid path characters.

File: src-tauri/src/commands/chat/send.rs [hunk 9]
```diff
@@ -2084,6 +2092,23 @@ pub async fn send_chat_message(
...
+                        let _ = db.clear_chat_session_state(&chat_session_id);
+                        let mut agents = state.agents.write().await;
+                        if let Some(session) = agents.get_mut(&chat_session_id) {
+                            session.session_id = String::new();
+                            session.turn_count = 0;
+                        }
+                        drop(agents);
+                        let e2 = crate::missing_cli::handle_err(&app, &e2).unwrap_or(e2);
+                        return Err(e2);
+                    }
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src-tauri/src/commands/chat/send.rs:2150]: Addressed in 0x72dd09. Extracted `fail_after_clearing_session_state` and replaced all four duplicated arms (two in the respawn path, two in the initial-start path) with a call into it. The helper opens a fresh `Database` connection internally because `rusqlite::Connection` is `!Sync` and holding a `&Database` across the agents-lock `await` would make the outer Tauri command future `!Send` — that matches the per-command DB pattern documented in CLAUDE.md. Doc comment on the helper explicitly calls out that diverging resets here previously caused `claude --session-id <stale-sid>` to re-launch fresh sessions pinned to dead transcripts, so future readers see the asymmetry that the helper exists to prevent.
